code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/EdgePersonalizationResponseHandler.java

```diff
@@ -261,92 +354,160 @@ void retrieveMessages(final List<Surface> surfaces, final Event event) {
     /**
      * Validates that the edge response event is a response that we are waiting for. If the returned payload is empty then the Messaging cache
      * and any loaded rules in the Messaging extension's {@link LaunchRulesEngine} are cleared.
-     * Non-empty payloads are converted into rules within {@link #parsePropositions(List, List, boolean, boolean)}.
      *
      * @param edgeResponseEvent A {@link Event} containing the in-app message definitions retrieved via the Edge extension.
      */
     void handleEdgePersonalizationNotification(final Event edgeResponseEvent) {
+        // validate this is one of our events
         final String requestEventId = MessagingUtils.getRequestEventId(edgeResponseEvent);
-
-        if (!messagesRequestEventId.equals(requestEventId) && !requestEventId.equals("TESTING_ID")) {
+        if (!requestedSurfacesForEventId.containsKey(requestEventId) && !requestEventId.equals("TESTING_ID")) {
```

Comment:
Should we add a null check for `requestEventId` here and fail fast instead of in `updateInProgressPropositionsWithEvent`?

Reply:
good point, i've added a fast fail for a null or empty request event id.